pkg/signup/service/signup_service_test.go

```diff
@@ -98,32 +130,50 @@ func (s *TestSignupServiceSuite) TestCreateUserSignup() {
 	ctx.Set(context.GivenNameKey, "jane")
 	ctx.Set(context.FamilyNameKey, "doe")
 	ctx.Set(context.CompanyKey, "red hat")
-	userSignup, err := s.Application.SignupService().CreateUserSignup(ctx)
-	require.NoError(s.T(), err)
-	require.NotNil(s.T(), userSignup)
 
-	gvk, err := apiutil.GVKForObject(userSignup, s.FakeUserSignupClient.Scheme)
-	require.NoError(s.T(), err)
-	gvr, _ := meta.UnsafeGuessKindToResource(gvk)
+	// when
+	userSignup, err := s.Application.SignupService().Signup(ctx)
 
-	values, err := s.FakeUserSignupClient.Tracker.List(gvr, gvk, s.Config().GetNamespace())
+	// then
 	require.NoError(s.T(), err)
+	gvr, existing := assertUserSignupExists(userSignup, userID.String())
 
-	userSignups := values.(*v1alpha1.UserSignupList)
-	require.NotEmpty(s.T(), userSignups.Items)
-	require.Len(s.T(), userSignups.Items, 1)
+	s.Run("deactivate and reactivate again", func() {
+		// given
+		deactivatedUS := existing
+		deactivatedUS.Spec.Deactivated = true
+		deactivatedUS.Status.Conditions = deactivated()
+		err := s.FakeUserSignupClient.Tracker.Update(gvr, &deactivatedUS, s.Config().GetNamespace())
```

Comment:
It's probably just a detail in this case, but it's safer to use the deep copy of the object:
```suggestion
		deactivatedUS := existing.DeepCopy()
		deactivatedUS.Spec.Deactivated = true
		deactivatedUS.Status.Conditions = deactivated()
		err := s.FakeUserSignupClient.Tracker.Update(gvr, deactivatedUS, s.Config().GetNamespace())
```

Reply:
Right. DeepCopy would be safer.